### DISCUSSION TITLE
How to reset server component if got error during fetch?

### DISCUSSION THREAD

**Discussion by ggukggukee** [vercel/next.js#64234] [App Router]
How to reset server component if there is an error during fetch?

```js
export default async function MainLayout({
  children,
}: {
  children: React.ReactNode;
}) {
  return (
    <>
      <Nav />
      <section className='flex gap-4 mt-4 px-4'>
        <CategoryList />
        {children}
        <section className='w-80 shrink-0 bg-background h-[calc(100vh-64px-16px-16px)] rounded-lg'>
          <Cart />
        </section>
      </section>
    </>
  );
}
```

I have fetch of categories inside `CategoryList` server component and want to show error component with reset button when i got error while fetching. How to do that?

**Comment by edspencer** [comment]
Not trying to self-promote, but I was wondering the same thing, did some research, figured it out and wrote it up on my blog at https://edspencer.net/2024/7/16/errors-and-retry-with-react-server-components - sounds like it may be what you're looking for.
